### ISSUE TITLE
[BUG] Compiling with LLVM-based Intel MPI compilers (oneAPI)

### ISSUE THREAD

**Issue by MarDiehl** [HDFGroup/hdf5#2186]
**Describe the bug**
`configure` stops with error. It is invoked with
```
/configure CC="mpiicc -cc=icx" CXX="mpiicpc -cxx=icpx"  F90="mpiifort -fc=ifx" FC="mpiifort -fc=ifx" F77="mpiifort -fc=ifx" --enable-shared --enable-parallel --enable-fortran --with-zlib=yes --with-szlib=no
```
Log is attached: [config.log](https://github.com/HDFGroup/hdf5/files/9863849/config.log)

**Expected behavior**
Installation with Intel oneAPI LLVM compilers should be possible

**Platform (please complete the following information)**
 - HDF5 1.12.2
 - Arch Linux (Kernel 6.0.2-arch1-1)
 - Intel oneAPI 2022.3.0.8767 (BaseKit + HPCKit)
 - Gnu Make 4.3
 - See above
 - IntelMPI (oneAPI)

**Additional context**
It seems that `configure` finds the classic intel compilers
```
...
compiler 'mpiicc -cc=icx' is Intel icc-2022.2.0
compiler 'mpiifort -fc=ifx' is Intel ifort-2022.2.0
compiler 'mpiicpc -cxx=icpx' is Intel icpc-2022.2.0
...
```

**Comment by kloeffel** [comment]
This error is known and documented. The error unfortunately comes from Autoconf <2.69. You can fix it by running autoreconf => 2.70 in the source directory.

https://www.intel.com/content/www/us/en/developer/articles/release-notes/oneapi-fortran-compiler-release-notes.html#inpage-nav-10-2-9

**Comment by MarDiehl** [comment]
@kloeffel many thanks!
